File: client/src/app/restaurant.service.ts
import { HttpClient } from "@angular/common/http"
import { inject, Injectable } from "@angular/core"
import { menuItem } from "./models"
import { Observable } from "rxjs"

@Injectable()
export class RestaurantService{ 

  private http = inject(HttpClient)
  
  // TODO: Task 2.2
  // You change the method's signature but not the name
  getMenu(): Observable<menuItem[]> {
    return this.http.get<menuItem[]>('/api/menu')
  }
  
  // TODO: Task 3.2
}

File: server/src/main/java/vttp/batch5/csf/assessment/server/services/RestaurantService.java
package vttp.batch5.csf.assessment.server.services;

import java.util.Collections;
import java.util.Comparator;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import vttp.batch5.csf.assessment.server.models.MenuItem;
import vttp.batch5.csf.assessment.server.repositories.OrdersRepository;

@Service
public class RestaurantService {

  @Autowired
  private OrdersRepository ordersRepo;


  // TODO: Task 2.2
  // You may change the method's signature
  public List<MenuItem> getMenu() {
    List<MenuItem> menus = ordersRepo.getMenu();
    Collections.sort(menus, Comparator.comparing(MenuItem::getName));
    return menus;
    
  }
  
  // TODO: Task 4


}
